pkg/trait/knative.go

```diff
@@ -101,6 +106,11 @@ func (t *knativeTrait) Configure(e *Environment) (bool, error) {
 
 			t.EndpointSinks = strings.Join(items, ",")
 		}
+		if t.FilterSourceChannels == nil && len(strings.Split(t.ChannelSources, ",")) > 1 {
```

Comment:
I think this should be an `or` because in case `FilterSourceChannels` is `false` but the number of sources is greater than two, then each `from` would receive a message for every subscribed channel